cmd/branch-planner/flags.go

```diff
@@ -61,5 +68,12 @@ func parseFlags() *applicationOptions {
 		opts.watchNamespace = opts.runtimeNamespace
 	}
 
+	// as in ../manager/main.go, this is the case where --no-cross-namespace-refs can be different and not overridden.
```

Comment:
```
if
      (allow-cross-namespace-refs changed)
  and (no-cross-namespace-refs changed)
then
  allow-cross-namespace-refs = true
  no-cross-namespace-refs = true
  opts.noCrossNamespaceRefs := false
```

So `allow-cross-namespace-refs` overrides `no-cross-namespace-refs`. This comment implies `--no-cross-namespace-refs` can't be overridden.

Why not value check instead of checking if the value changed? usually I don't think I have issues with logical constructs, but I had to write notes to understand all cases properly as the "changed" is not the same as "if it's true". In this case technically yes, as both has `false` as default value, but still.

Reply:
> Why not value check instead of checking if the value changed?

```
	// --allow... \ --no... | true | false |  (false) |
	// ---------------------|------|-------|----|
	//     true             |  t   |   t   |  t |
	//     false            |  f   |   f   |  f |
	//     (false)          |  f   |   t*  |  f |
```

This is the table from cmd/manager/main.go, with the defaults in parentheses. The desired outcome for `--allow-cross-namespace-refs=false --no-cross-namespace-refs=false` is to disallow cross-namespace refs; but the desired outcome for `--no-cross-namespace-refs=false`, for which the flag values are the same as the previous case, is to allow cross-namespace refs.

It is annoyingly complicated. I think.

Reply:
But what is the desired outcome if both set to true? As I understand it will disable `noCrossNamespaceRefs`, while the comment itself indicates (as I read) it should be `noCrossNamespaceRefs = true` because `noCrossNamespaceRefs` is explicitly set to `true` and it says "--no-cross-namespace-refs can be different and not overridden"

Reply:
If `--allow-cross-namespace-refs --no-cross-namespace-refs` the desired outcome is to allow cross-namespace refs (topmost, leftmost cell in the table). In this case, the conditional on L72 will evaluate to false and L75 will use the value from allowCrossNamespaceRefs, i.e., it will allow cross-namespace refs. So we agree on the desired outcome, and the code agrees too.

On the comment -- "this is the case where --no-cross-namespace-refs can be different and not overridden" -- this refers to the conditional checking that --no-cross-namespaces-refs is supplied but --allow-etc. is not (--> "not overridden"), giving a set of outcomes that differs to the other columns in the table. It's not a very useful comment. I'll change it.